resources/packs/gcp/kms.go

```diff
@@ -0,0 +1,347 @@
+	if err != nil {
+		return nil, err
+	}
+
+	return g.MotorRuntime.CreateResource("gcp.project.kmsservices",
+		"projectId", projectId,
+	)
+}
+
+func (g *mqlGcpProjectKmsservicesKeyring) id() (string, error) {
+	return g.Id()
+}
+
+func (g *mqlGcpProjectKmsservicesKeyringCryptokey) id() (string, error) {
+	return g.Id()
+}
+
+func (g *mqlGcpProjectKmsservicesKeyringCryptokeyVersion) id() (string, error) {
+	return g.Id()
+}
+
+func (g *mqlGcpProjectKmsservicesKeyringCryptokeyVersionAttestation) id() (string, error) {
+	return g.Id()
+}
+
+func (g *mqlGcpProjectKmsservicesKeyringCryptokeyVersionExternalProtectionLevelOptions) id() (string, error) {
+	return g.Id()
+}
+
+func (g *mqlGcpProjectKmsservicesKeyringCryptokeyVersionAttestationCertificatechains) id() (string, error) {
+	return g.Id()
+}
+
+func (g *mqlGcpProjectKmsservices) GetLocations() ([]interface{}, error) {
+	projectId, err := g.ProjectId()
+	if err != nil {
+		return nil, err
+	}
+
+	provider, err := gcpProvider(g.MotorRuntime.Motor.Provider)
+	if err != nil {
+		return nil, err
+	}
+
+	creds, err := provider.Credentials(kms.DefaultAuthScopes()...)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx := context.Background()
+
+	kmsSvc, err := kms.NewKeyManagementClient(ctx, option.WithCredentials(creds))
+	if err != nil {
+		return nil, err
+	}
+	defer kmsSvc.Close()
+
+	var locations []interface{}
+	it := kmsSvc.ListLocations(ctx, &location.ListLocationsRequest{Name: fmt.Sprintf("projects/%s", projectId)})
+	for {
+		l, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		locations = append(locations, l.LocationId)
+	}
+	return locations, nil
+}
+
+func (g *mqlGcpProjectKmsservices) GetKeyrings() ([]interface{}, error) {
+	projectId, err := g.ProjectId()
+	if err != nil {
+		return nil, err
+	}
+
+	locations, err := g.Locations()
+	if err != nil {
+		return nil, err
+	}
+
+	provider, err := gcpProvider(g.MotorRuntime.Motor.Provider)
+	if err != nil {
+		return nil, err
+	}
+
+	creds, err := provider.Credentials(kms.DefaultAuthScopes()...)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx := context.Background()
+
+	kmsSvc, err := kms.NewKeyManagementClient(ctx, option.WithCredentials(creds))
+	if err != nil {
+		return nil, err
+	}
+	defer kmsSvc.Close()
+
+	var keyrings []interface{}
+	var wg sync.WaitGroup
+	wg.Add(len(locations))
+	mux := &sync.Mutex{}
+
+	for _, location := range locations {
+		go func(svc *kms.KeyManagementClient, project string, location string) {
+			defer wg.Done()
+			it := kmsSvc.ListKeyRings(ctx,
+				&kmspb.ListKeyRingsRequest{Parent: fmt.Sprintf("projects/%s/locations/%s", projectId, location)})
+			for {
+				k, err := it.Next()
+				if err == iterator.Done {
+					break
+				}
+				if err != nil {
+					log.Error().Err(err)
+					return
+				}
+
+				created := k.CreateTime.AsTime()
+				mqlKeyring, err := g.MotorRuntime.CreateResource("gcp.project.kmsservices.keyring",
+					"id", k.Name,
```

Comment:
lets remove the id field and change the id function to use name as default. Same applies to the other resources